sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2254,6 +2254,14 @@ object SQLConf {
    * The map contains info about removed SQL configs. Keys are SQL config names,
    * map values contain extra information like the version in which the config was removed,
    * config's default value and a comment.
+   *
+   * Removed SQL config should be added to the map iff its non-default value changed
+   * the behavior. Any removed config of `removedSQLConfigs` which set to a non-default
+   * value raises an exception propagated to user's app. For example, the SQL config
+   * `spark.sql.variable.substitute.depth` was deprecated since Spark 2.1 and not
+   * used in Spark 2.4, so, it should not be placed here because any of its values
+   * does not impact on the behavior of Spark 3.0. And this makes migrations to new
```

Comment:
Shall we describe in a simpler and positive words like `because the config is no-op technically` instead of `because any of its values does not impact on the behavior of Spark`?

Reply:
Yeah, I was about to point this out. Let's make the description shorter and simpler.